index.js

```diff
@@ -1,12 +1,122 @@
 const express = require('express');
 const helmet = require('helmet');
+const knex = require('knex');
+
+const knexConfig = {
+  client: 'sqlite3',
+  connection: {
+    filename: './data/lambda.sqlite3'
+  },
+  useNullAsDefault: true
+  // debug: true
+};
+
+const db = knex(knexConfig);
 
 const server = express();
 
 server.use(express.json());
 server.use(helmet());
 
-// endpoints here
+server.post('/api/zoos', (req, res) => {
+  const { name } = req.body;
+  if (!name) {
+    res
+      .status(400)
+      .json({ error: 'You must provide a name for the new zoo record.' });
+  } else {
+    db('zoos')
+      .insert({ name })
+      .then(ids => {
+        const id = ids[0];
+        db('zoos')
+          .where({ id })
+          .first()
+          .then(zoo => {
+            res.status(200).json(zoo);
+          });
+      })
+      .catch(err =>
+        res.status(500).json({
+          // I tried adding another catch after the .then() where the added record is retrieved, but when I intentionally caused an error in the retrieval (after successully adding the record), it displayed this error message anyway (so I rewrote this message to acknowledge both possible causes).
+          error: `There was an error adding the zoo data, or an error retrieving the added zoo data. ${err}`
+        })
+      );
+  }
+});
+
+server.get('/api/zoos', (req, res) => {
+  db('zoos')
+    .then(zoos => {
+      res.status(200).json(zoos);
+    })
+    .catch(err =>
+      res.status(500).json({
+        error: `There was an error retrieving the zoos data. ${err}`
+      })
+    );
+});
+
+server.get('/api/zoos/:id', (req, res) => {
+  db('zoos')
+    .where('id', req.params.id)
```

Comment:
you can shorten this up if you first destructure the id off the params coming in (const {id} = req.params) since this `where` statement can then be mutated to `where('id', id)` and then as an object `where({id: id})` and _then_ the extra-short equal-property/value reference ES6 syntax of `where({id})` since the variable you assign to the property has the same name